templates/common-templates.adoc

```diff
@@ -174,3 +173,207 @@ Please note that, after the generation step, VM objects and template
+`defaults.template.cnv.io`, which is omitted below for brevity. So the
+actual annotations you should use will look like
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Template
+metadata:
+  name: windows-10
+  annotations:
+    defaults.template.cnv.io/disk: default-disk
+    defaults.template.cnv.io/volume: default-volume
+    defaults.template.cnv.io/nic: default-nic
+    defaults.template.cnv.io/network: default-network
+----
+
+Unless otherwise specified, all annotations are meant to be safe
+defaults, both for performance and compability, and hints for the
+CNV-aware UI and tooling.
+
+disk
+++++
+
+See the section `references` below.
+
+Example:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Template
+metadata:
+  name: Linux
+  annotations:
+    defaults.template.cnv.io/disk: rhel-disk
+----
+
+nic
++++
+
+See the section `references` below.
+
+Example:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Template
+metadata:
+  name: Windows
+  annotations:
+    defaults.template.cnv.io/nic: my-nic
+----
+
+volume
+++++++
+
+See the section `references` below.
+
+Example:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Template
+metadata:
+  name: Linux
+  annotations:
+    defaults.template.cnv.io/volume: custom-volume
+----
+
+network
++++++++
+
+See the section `references` below.
+
+Example:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Template
+metadata:
+  name: Linux
+  annotations:
+    defaults.template.cnv.io/network: fast-net
+----
+
+references
+++++++++++
+
+The default values for network, nic, volume, disk is meant to be the *name* of a section later in the document that the UI will find and consume to find the default values for the corresponding types.
+For example, considering the annotation `defaults.template.cnv.io/disk: my-disk`: we assume that later in the document it exists an element called `my-disk` that the UI can use to find the data it needs.
+The names actually don't matter as long as they are legal for kubernetes and consistent with the content of the document.
+
+complete example
+++++++++++++++++
+
+[demo-template.yaml]
+----
+apiversion: v1
+items:
+- apiversion: kubevirt.io/v1alpha3
+  kind: virtualmachine
+  metadata:
+    labels:
+      vm.cnv.io/template: rhel7-generic-tiny
+    name: rheltinyvm
+    osinfoname: rhel7.0
+    defaults.template.cnv.io/disk: rhel-default-disk
+    defaults.template.cnv.io/nic: rhel-default-nic
+  spec:
+    running: false
+    template:
+      spec:
+        domain:
+          cpu:
+            sockets: 1
+            cores: 1
+            threads: 1
+          devices:
+            rng: {}
+          resources:
+            requests:
+              memory: 1g
+        terminationgraceperiodseconds: 0
+        volumes:
+        - containerDisk:
+          image: registry:5000/kubevirt/cirros-container-disk-demo:devel
+          name: rhel-default-disk
+        networks:
+        - genie:
+          networkName: flannel
+        name: rhel-default-nic
```

Comment:
The indentation looks wrong and also you use NIC to reference network.

Reply:
Right, fixed